eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -18,8 +18,8 @@ jobs:
       variables:
         - template: /eng/pipelines/templates/variables/globals.yml
       pool:
-        name: "azsdk-pool-mms-ubuntu-2004-general"
-        vmImage: "MMSUbuntu20.04"
+        name: "azsdk-pool"
+        vmImage: "ubuntu-24.04"
```

Comment:
I'm not sure if this works, or needs to be converted to `demands:`.

Reply:
I think this needs to be converted to a demands.

Reply:
@jeremymeng: How can we test this?

Reply:
@mikeharder I don't think we are running smoke tests at the moment after https://github.com/Azure/azure-sdk-for-js/pull/24430. We can probably skip testing smoke-tests for now.

Reply:
Anything else in this PR you think should be fixed before this is merged?  Otherwise I'm fine with merging as-is, and I can fast fix anything that breaks in prod.